lib/improver/weights.py

```diff
@@ -451,3 +451,110 @@ def __repr__(self):
         desc = ('<ChooseDefaultWeightsNonLinear '
                 'cval={0:4.1f}>'.format(self.cval))
         return desc
+
+
+class ChooseDefaultWeightsTriangular(object):
+    """ Calculate Default Weights using a Triangular Function. """
+    def __init__(self, width, midpoint, units="no_unit"):
```

Comment:
As `no_unit` seems to have a specific meaning within the cf_units module, I think it would be acceptable to leave this as `no_unit` here.